### ISSUE TITLE
Snap to Grid every element in the page

### ISSUE THREAD

**Issue by R4ygen** [stevencohn/OneMore#1603]
# Problem to Solve

So, I use OneNote with the "Snap to Grid" option always on. This permits me to take well organized notes, and helps me to respect the actual background grid of the pages, automatically, without the need to always manually adjust the position of text boxes, shapes, or images.
![image](https://github.com/user-attachments/assets/bed29b8f-0f64-42cb-8721-1f6d4c63dbae)

Issue is: Most of the time I edit something, move a drawing, add space, or else... elements in the page, like text boxes, become misaligned from the grid. So when I then go to add a new text box, is not aligned with the rest of the page, because the whole page is not aligned anymore to the grid.
These are the small things in Onenote that makes me want to switch to another software really, but there are no alternatives like Onenote, so, I thought about making a feauture request here that fixes this issue instead of waiting on Microsoft to fix it.

# New Feature/Solution

A button, that when clicked, it snaps every element in the page (text boxes, shapes, pictures, etc...) to the nearest grid snap point to them.

## Alternative Solutions

The fix I'm manually applying right now is to select everything on the page, but carefully de-select every drawing and shapes, because those do not snap to the grid (apart from shapes), and then move everything just a bit. It will naturally snap back to the page. This only works if every element in the page was already aligned to the grid in the first place though (aka, all the element are not misaligned relative to each other), otherwise you will have more than one stable snap positions possible for this whole selection you are moving. And, as said, it only works if you don't include any drawing in the selection, not even shapes (even though, those DO snap to grid by themselfes).

**Comment by stevencohn** [comment]
### Notes
* PageSettings/RuleLines indicates horizontal/spacing and vertical/spacing
* Outline/Position x, y attributes can be calculated to approximate closest grid intersection
   * Outline positions should be a multiple of spacing
* InkDrawing/Position x, y attributes can be calculated to approximate closest grid intersection

### Additional
* Adjust Grid to Text command could be updated to auto-align Outlines to adjust grid

### Out of scope
* Ignore background images
